crates/ruff_python_formatter/src/expression/binary_like.rs

```diff
@@ -667,7 +667,10 @@ impl Format<PyFormatContext<'_>> for FlatBinaryExpressionSlice<'_> {
                     leading_comments(leading).fmt(f)?;
                 }
 
-                in_parentheses_only_group(&left).fmt(f)?;
+                match &left.0 {
```

Comment:
This should reduce the number of groups that we write because we avoid grouping e.g. a single identifier `a` and we instead rely on expressions that can be grouped to create their own groups (which we were already doing except for subscript)